dll/win32/ucrtbase/printf.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * PROJECT:     ReactOS CRT
+ * LICENSE:     MIT (https://spdx.org/licenses/MIT)
+ * PURPOSE:     Instantiation of CRT stdio inline functions
+ * COPYRIGHT:   Copyright 2023-2024 Timo Kreuzer <timo.kreuzer@reactos.org>
+ */
+
+#define _NO_CRT_STDIO_INLINE
+#include <stdio.h>
+
+// These 2 functions are inlined in UCRT headers, but they are referenced by
+// external code in the legacy CRT, so we need to provide them as non-inline
+// functions here.
```

Comment:
Couldn't you include that header here?

Reply:
On MSVC `_CRT_STDIO_INLINE` is defined to `__inline`, which will generate globally visible symbols, but only when the function is referenced somewhere. With MSVC we could pull them in by doing something like `void *_p_sprintf = (void*)sprintf;`
But GCC handles inline functions differently: if they are not static, they will conflict with each other, when they are used in different compilation units, so they have to be declared `static __inline`, which means no symbols are created. We could define `_CRT_STDIO_INLINE` to nothing here, before including stdio.h, but that would create global functions for everything, which we don't want.

Reply:
sad, but not entirely unexpected since we have to support gcc.